lib/yeti_exception/error.rb

```diff
@@ -0,0 +1,60 @@
+require 'json'
+
+module YetiException
+
+  # Enhanced exception class, with some features added beyond those in
+  # StandardError:
+  #
+  # - The class in which the exception was raised.  This can be useful for
+  #   logging, since the backtrace is much more verbose.
+  #
+  # - A details hash, containing arbitrary key/value pairs related to the
+  #   exception.
+  #
+  # - A transient flag, defaulting to true, indicating whether the exception
+  #   should be considered transient or final.  This can be used when rescuing
+  #   the exception to determine whether the operation should be retried.
+  #
+  # - An HTTP status, defaulting to 500.  This can be used to generate the
+  #   response payload if the exception is rescued as part of an HTTP request.
+  #
+  # - An optional static message, definable by subclasses, to be included
+  #   automatically in the details hash.  This removes the need to pass this
+  #   message in explicitly during initialization.
+
+  class Error < StandardError
```

Comment:
[nit] If I hadn't seen this PR, I would expect a class named `Error` to be part of core ruby. Might we want to name this something like `YetiError` so that it's more clear to developers using it that this is Yesware's custom error class?

Reply:
It's in the `YetiException` namespace.

Reply:
Is that how it will appear in logs and code? Always as `YetiException::Error` and never as plain `Error`?

Reply:
Correct